Jenkinsfile

```diff
@@ -601,47 +629,54 @@ pipeline {
                         return PIPELINE_CONTROL.ci_skip == false
                     }
                     expression {
-                        return PIPELINE_CONTROL.deploy
+                        return PIPELINE_CONTROL.cd_skip == false
                     }
                     expression {
                         return RELEASE_BRANCHES.contains(BRANCH_NAME)
                     }
-                    expression {
-                        return GIT_COMMIT != GIT_PREVIOUS_SUCCESSFUL_COMMIT
-                    }
                 }
             }
             steps {
                 timeout(time: 5, unit: 'MINUTES') {
                     echo 'Deploy Binary'
-                    withCredentials([usernamePassword(credentialsId: ARTIFACTORY_CREDENTIALS_ID, usernameVariable: 'USERNAME', passwordVariable: 'API_KEY')]) {
-
-                        // Set up authentication to Artifactory
+                    script {
                         sh "rm -f .npmrc"
-                        sh 'curl -u $USERNAME:$API_KEY https://eu.artifactory.swg-devops.com/artifactory/api/npm/auth/ >> .npmrc'
-                        sh "echo registry=$TEST_NPM_REGISTRY >> .npmrc"
+                        // set @zowe registry
+                        sh "echo always-auth=true >> .npmrc"
                         sh "echo @brightside:registry=https://api.bintray.com/npm/ca/brightside/ >> .npmrc"
                         sh "echo @brightside:always-auth=false >> .npmrc"
-                        
-                        script {
-                            if (BRANCH_NAME == MASTER_BRANCH) {
-                                echo "publishing next to $TEST_NPM_REGISTRY"
-                                sh "npm publish --tag alpha"
-                            } else if (BRANCH_NAME == DEV_BRANCH) {
-                                echo "publishing next to $TEST_NPM_REGISTRY"
-                                sh "npm publish --tag next"
-                            } else {
-                                echo "publishing latest to $TEST_NPM_REGISTRY"
-                                sh "npm publish --tag latest"
+                        if (BRANCH_NAME == MASTER_BRANCH) {
+                            // Set credential
+                            sh "echo _auth=${NPM_CREDENTIALS} >> .npmrc"
```

Comment:
I think the e-mail address will be needed in `.npmrc` as well

Reply:
which email address should I use?

Reply:
The credential is set up in Jenkins as a username and password so I think you should be able to get the email address with `$NPM_CREDENTIALS_USR` and the password with `$NPM_CREDENTIALS_PWD` as described here: https://jenkins.io/doc/book/pipeline/jenkinsfile/#usernames-and-passwords
Also I think you could avoid writing the

Reply:
In fact email address is not needed as the token has the email information, however I changed NPM_CREDENTIALS to NPM_CREDENTIALS_PSW as this is the correct token.